### PR TITLE
WT-2932 Add a configuration option allowing tables to ignore cache limits

### PR DESCRIPTION
**Pull request by agorrod**
Only available for in-memory configurations, designed to be used for
applications that maintain a small amount of content that is required
for correctness.

### CODE REVIEW

**Comment by agorrod** [comment]
autobuild: retest this please

File: src/include/api.h [hunk 1]
```diff
@@ -139,7 +139,9 @@
 	(s) = (WT_SESSION_IMPL *)(cur)->session;			\
 	TXN_API_CALL_NOCONF(s, WT_CURSOR, n, cur,			\
 	    ((bt) == NULL) ? NULL : ((WT_BTREE *)(bt))->dhandle);	\
-	if (F_ISSET(S2C(s), WT_CONN_IN_MEMORY) && __wt_cache_full(s))	\
+	if (F_ISSET(S2C(s), WT_CONN_IN_MEMORY) &&			\
+	    !F_ISSET((WT_BTREE *)(bt), WT_BTREE_IGNORE_CACHE) &&	\
+	    __wt_cache_full(s))						\
```
Comment (michaelcahill) [inline src/include/api.h]: whitespace before backslash

Comment (agorrod) [inline src/include/api.h]: I believe that is a git diff-ism, there is a tab character after the ``&``

File: dist/api_data.py [hunk 2]
```diff
@@ -183,6 +183,12 @@ def __cmp__(self, other):
         configure Huffman encoding for values.  Permitted values are
         \c "none", \c "english", \c "utf8<file>" or \c "utf16<file>".
         See @ref huffman for more information'''),
+    Config('in_memory_ignore_cache_size', 'false', r'''
+        allow update and insert operations to proceed even if the cache is
+        already at capacity. Only valid in conjunction with in-memory
+        databases. Should be used with caution - this configuration allows
+        WiredTiger to consume memory over the configured cache limit''',
+        type='boolean'),
```
Comment (michaelcahill) [inline dist/api_data.py]: How about renaming this `ignore_in_memory_cache_size`?  I don't feel super strongly, but having the verb at the beginning reads better to me.

File: src/include/api.h (hunk 1 above)
Comment (michaelcahill) [inline src/include/api.h]: OK, that's an odd one.

**Pull request merged**